File: mysql_examples/recipes/metadata/get_storage_engines.pl
#!/usr/bin/perl
# get_storage_engines.pl - list supported storage engines

use strict;
use warnings;
use Cookbook;

#@ _GET_STORAGE_ENGINES_
sub get_storage_engines
{
my $dbh = shift;
my @engines;

  my $sth = $dbh->prepare ("SHOW ENGINES");
  $sth->execute ();
  while (my @row = $sth->fetchrow_array ())
  {
    # $row[0] = engine name, $row[1] = support
    $row[1] = uc ($row[1]);   # uppercase the "support" value
    push (@engines, $row[0]) if $row[1] eq "YES" || $row[1] eq "DEFAULT";
  }
  return @engines;
}
#@ _GET_STORAGE_ENGINES_

my $dbh = Cookbook::connect ();

my @engines = get_storage_engines ($dbh);
print "Supported storage engines:\n";
print join ("\n", @engines), "\n";

$dbh->disconnect ();

File: mysql_examples/recipes/metadata/get_storage_engines.rb
#!/usr/bin/ruby -w
# get_storage_engines.rb - list supported storage engines

require "Cookbook"

#@ _GET_STORAGE_ENGINES_
def get_storage_engines(dbh)
  engines = []
  dbh.select_all("SHOW ENGINES").each do |engine, support|
    engines << engine if ["YES", "DEFAULT"].include?(support.upcase)
  end
  return engines
end
#@ _GET_STORAGE_ENGINES_

begin
  dbh = Cookbook.connect
rescue DBI::DatabaseError => e
  puts "Could not connect to server"
  puts "Error code: #{e.err}"
  puts "Error message: #{e.errstr}"
end

engines = get_storage_engines(dbh)
puts "Supported storage engines:"
engines.each do |engine| puts engine end

dbh.disconnect
